website/files/models/github.py

```diff
@@ -14,3 +14,7 @@ class GithubFolder(GithubFileNode, Folder):
 
 class GithubFile(GithubFileNode, File):
     version_identifier = 'ref'
+
+    def touch(self, auth_header, revision=None, ref=None, branch=None, **kwargs):
+        revision = revision or ref or branch
```

Comment:
@abought the order here is intentional and correct.
We plan on moving towards using just revision at some point so that should always take priority. 
A git `ref` is either a branch, commit sha, or file sha, it is accepted over branch as it is more specific. IE I could be view a past file on a different branch, in which case both `ref` and `branch` could be present. Both would be correct but `ref` would be more correct.